### PR TITLE
Test cleanup and fixes

### PR DESCRIPTION
**Pull request by BrennanConroy**
Fixed a flaky test and made the in-process server use async

### CODE REVIEW

File: src/SignalR/clients/csharp/Client/test/FunctionalTests/HubConnectionTests.cs [hunk 1]
```diff
@@ -1626,10 +1627,10 @@ public async Task WebSocketOptionsAreApplied()
             }
         }
 
-        [Fact(Skip = "Returning object from Hub method not support by System.Text.Json yet")]
+        [Fact]
```
Comment (BrennanConroy) [inline src/SignalR/clients/csharp/Client/test/FunctionalTests/HubConnectionTests.cs]: unskipped a test

File: src/SignalR/common/Http.Connections/test/HttpConnectionDispatcherTests.cs [hunk 2]
```diff
@@ -1166,6 +1166,18 @@ public async Task RequestToActiveConnectionIdKillsPreviousConnectionLongPolling(
...
+                while (connection.TransportTask.IsCompleted && count < 50)
+                {
+                    count++;
+                    await Task.Delay(15);
+                }
+                if (count == 50)
+                {
+                    Assert.True(false, "Poll took too long to start");
+                }
                 Assert.Equal(HttpConnectionStatus.Active, connection.Status);
```
Comment (BrennanConroy) [inline src/SignalR/common/Http.Connections/test/HttpConnectionDispatcherTests.cs]: Deflaked a test

File: src/SignalR/clients/csharp/Client/test/FunctionalTests/HubConnectionTests.cs [hunk 3]
```diff
@@ -1640,10 +1641,10 @@ public async Task CheckHttpConnectionFeatures()
                     await hubConnection.StartAsync().OrTimeout();
 
                     var features = await hubConnection.InvokeAsync<object[]>(nameof(TestHub.GetIHttpConnectionFeatureProperties)).OrTimeout();
```
Comment (halter73) [inline src/SignalR/clients/csharp/Client/test/FunctionalTests/HubConnectionTests.cs]: Nit: would the following avoid the casts?

```suggestion
                    var features = await hubConnection.InvokeAsync<JsonElement[]>(nameof(TestHub.GetIHttpConnectionFeatureProperties)).OrTimeout();
```

File: src/SignalR/common/Http.Connections/test/HttpConnectionDispatcherTests.cs [hunk 4]
```diff
@@ -1166,6 +1166,18 @@ public async Task RequestToActiveConnectionIdKillsPreviousConnectionLongPolling(
                 await request1.OrTimeout();
 
                 Assert.Equal(StatusCodes.Status204NoContent, context1.Response.StatusCode);
+
+                count = 0;
+                // Wait until the second request has started internally
+                while (connection.TransportTask.IsCompleted && count < 50)
```
Comment (halter73) [inline src/SignalR/common/Http.Connections/test/HttpConnectionDispatcherTests.cs]: Why not just await the connection.TransportTask with a timeout? Is the connection.TransportTask changing from underneath you?

**Review by halter73 — APPROVED** [review]

File: src/SignalR/common/Http.Connections/test/HttpConnectionDispatcherTests.cs (hunk 4 above)
Comment (BrennanConroy) [reply src/SignalR/common/Http.Connections/test/HttpConnectionDispatcherTests.cs]: It's going to be completed from the first poll request, so we're waiting until it's been set by the second request.

Comment (halter73) [reply src/SignalR/common/Http.Connections/test/HttpConnectionDispatcherTests.cs]: Ahh. I misread. I though this was looping while it wasn't completed, not looping while it is completed.

**Pull request merged**